### ISSUE TITLE
Question about block diffusion attention mask

### ISSUE THREAD

**Issue by weirayao** [kuleshov-group/bd3lms#27]
Hi authors, thanks for the great work! Wondering why we need the attention weights for the yellow block? Are we also reconstructing the unmasked tokens with unmasked tokens? My understanding is the loss only includes reconstructing masked tokens, with the current block and ground truth tokens in past block.

<img width="1073" alt="Image" src="https://github.com/user-attachments/assets/ca79b59d-098d-4424-bfa7-7c26ce6d6e94" />

**Comment by mariannearr** [comment]
Thank you for your interest!

Since each tokens in each noised block cross-attend to the activations of previous clean blocks (blue cells), we also compute their activations in every forward pass (yellow cells).

I hope this resolves your question.

**Comment by haotiansun14** [comment]
> Thank you for your interest!
> 
> Since each tokens in each noised block cross-attend to the activations of previous clean blocks (blue cells), we also compute their activations in every forward pass (yellow cells).
> 
> I hope this resolves your question.

Hi,

Thanks for this excellent work! Could you please elaborate more on the purpose of M_BC? It seems combining M_BD and M_OBC is sufficient for training (the model attends to previous blocks w/ clean data and current block w/ noisy data).
